audio_service/android/src/main/java/com/ryanheise/audioservice/AudioServicePlugin.java

```diff
@@ -283,62 +288,145 @@ public void onDetachedFromActivity() {
         activityPluginBinding = null;
         newIntentListener = null;
         clientInterface.setActivity(null);
-        clientInterface.setContext(flutterPluginBinding.getApplicationContext());
-        if (clientInterfaces.size() == 1) {
-            // This unbinds from the service allowing AudioService.onDestroy to
-            // happen which in turn allows the FlutterEngine to be destroyed.
-            disconnect();
-        }
-        if (clientInterface == mainClientInterface) {
-            mainClientInterface = null;
-        }
+        disconnect();
+    }
+
+    private void registerOnNewIntentListener() {
+        activityPluginBinding.addOnNewIntentListener(newIntentListener = (intent) -> {
+            clientInterface.activity.setIntent(intent);
+            return true;
+        });
     }
 
+    /** Connects to the service. */
     private void connect() {
-        /* Activity activity = mainClientInterface.activity; */
-        /* if (activity != null) { */
-        /*     if (clientInterface.wasLaunchedFromRecents()) { */
-        /*         // We do this to avoid using the old intent. */
-        /*         activity.setIntent(new Intent(Intent.ACTION_MAIN)); */
-        /*     } */
-        /*     if (activity.getIntent().getAction() != null) */
-        /*         invokeClientMethod("notificationClicked", activity.getIntent().getAction().equals(AudioService.NOTIFICATION_CLICK_ACTION)); */
-        /* } */
+        // Activity activity = clientInterface.activity;
+        // boolean launchedFromRecents = (activity.getIntent().getFlags() & Intent.FLAG_ACTIVITY_LAUNCHED_FROM_HISTORY) ==
+        //         Intent.FLAG_ACTIVITY_LAUNCHED_FROM_HISTORY;
+        // if (!launchedFromRecents) {
+        //     invokeClientMethod("notificationClicked", activity.getIntent().getAction().equals(AudioService.NOTIFICATION_CLICK_ACTION));
+        // }
         if (mediaBrowser == null) {
-            mediaBrowser = new MediaBrowserCompat(applicationContext,
+            Context applicationContext = flutterPluginBinding.getApplicationContext();
+            mediaBrowser = new MediaBrowserCompat(
+                    applicationContext,
                     new ComponentName(applicationContext, AudioService.class),
-                    connectionCallback,
-                    null);
+                    connectionCallback(),
+                    null
+            );
             mediaBrowser.connect();
         }
     }
 
-    private void disconnect() {
-        Activity activity = mainClientInterface != null ? mainClientInterface.activity : null;
-        if (activity != null) {
-            // Since the activity enters paused state, we set the intent with ACTION_MAIN.
-            activity.setIntent(new Intent(Intent.ACTION_MAIN));
-        }
```

Comment:
I'm not really sure about the purpose of this code, other than that, chaning the activity intent from inside the plugin doesn't look good